raw/db.py

```diff
@@ -128,36 +128,67 @@ def result(sql, returns="dict", autocommit=False, **kwargs):
 
 def result_from_file(path, returns="dict", autocommit=False, **kwargs):
     """Read SQL from file at `path` and submit via `result()` method"""
-    pathobj = Path(path)
-    # If path doesn't exist
-    if not pathobj.exists():
-        raise IOError(f"File '{path}' not found!")
-
-    # If it's a directory
-    if pathobj.is_dir():
-        raise IOError(f"'{path}' is a directory!")
-
-    # Read the given file into memory and pass to result().
-    with open(path) as f:
-        sql = f.read()
-        rows = result(sql=sql, returns=returns, **kwargs)
-        return rows
+    # If `path` is a S3 url, read it from there
+    if path.startswith("s3://"):
+        bucket = s3fs.S3FileSystem(anon=False)
+        with bucket.open(path, "rt") as f:
```

Comment:
ℹ️ The documentation for s3fs says that [only binary](https://s3fs.readthedocs.io/en/latest/index.html#limitations) mode is currently supported here.

> modestr
>     One of ‘rb’, ‘wb’, ‘ab’. These have the same meaning as they do for the built-in open function.